File: II_semestr/programowanie_obiektowe(cpp)/zaj6/zaj6/kwadrat.cpp
#include "kwadrat.h"

#include <iostream>
#include <string.h>


using namespace std;


kwadrat::kwadrat()
{
    nazwa = "kwadrat_def";
    a = 2;
}


kwadrat::kwadrat(float a)
{
    nazwa = "kwadrat";
    this->a = a;
}

kwadrat::~kwadrat()
{

}

float kwadrat::pole()
{
    return a*a;
}

float kwadrat::obwod()
{
    return 4*a;
}

File: III_semestr/programowanie_obiektowe(java)/lab5/lab5/src/main/java/com/lab5/Kwadrat.java
package com.lab5;

import java.awt.Color;

public class Kwadrat extends Prostokat
{
    public

        void kwadrat(int a, int liczbaBokow, int liczbaWierzcholkow, int sumaKatowWewnetrznych, Color kolor)
        {
            this.a = a;
            this.liczbaBokow = liczbaBokow;
            this.sumaKatowWewnetrznych = sumaKatowWewnetrznych;
            setColor(kolor);
        }

        int obliczObwod()
        {
            this.obwod = a * liczbaBokow;
            return obwod;
        }

        int obliczPole()
        {
            this.pole = a * a;
            return pole;
        }
        

        @Override
        int getA()
        {
            return a;
        }

        @Override
        void setA(int a)
        {
            this.a = a;
        }
}
